### PR TITLE
Support GCE jwt validation

### PR DESCRIPTION
**Pull request by wattli**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by wattli** [comment]
The failed presubmit seems unrelated

File: security/pkg/platform/gcp.go [hunk 1]
```diff
@@ -91,8 +91,15 @@ func (ci *GcpClientImpl) GetDialOptions() ([]grpc.DialOption, error) {
 
 // GetServiceIdentity gets the identity of the GCE service.
 func (ci *GcpClientImpl) GetServiceIdentity() (string, error) {
-	// TODO(wattli): update this once we are ready for GCE
-	return "", nil
+	serviceAccount, err := ci.fetcher.FetchServiceAccount()
+	if err != nil {
+		glog.Errorf("Failed to get service account from GCE metadata %s, please make sure this binary is running on a GCE VM", err)
```
Comment (myidpt) [inline security/pkg/platform/gcp.go]: %v?

**Pull request merged**